pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -501,6 +501,23 @@ func (m *mgr) deleteNamespace(obj runtime.Object, controller string) error {
 	if apierrors.IsNotFound(err) {
 		return nil
 	}
+
+	sleep := 10         // sleep interval in seconds
+	pollDuration := 600 // Total poll duration in seconds
+	//  Polls to verify if the cluster exists every 10 seconds for a total duration of 10 minutes.
+	for i := 0; i < pollDuration/sleep; i++ {
+		c, err := m.mgmt.Wrangler.CAPI.Cluster().Get(o.GetName(), o.GetName(), v1.GetOptions{})
```

Comment:
can you use o.GetNamespace() for the name

Reply:
I tried using it but for some reason It has a null value.

Reply:
Instead I used ns.Name to make it meaningful. Pushed the changes.